src/emulator/pubsubEmulator.ts

```diff
@@ -62,7 +70,26 @@ export class PubsubEmulator implements EmulatorInstance {
   }
 
   async stop(): Promise<void> {
-    await downloadableEmulators.stop(Emulators.PUBSUB);
+    try {
+      await downloadableEmulators.stop(Emulators.PUBSUB);
+    } catch (e: unknown) {
+      this.logger.logLabeled("DEBUG", "pubsub", JSON.stringify(e));
+      if (process.platform !== "win32") {
+        // The exec is wrapped in a promise so we can block on it's completion via `await`.
+        const execPromise = new Promise((resolve) => {
+          exec(PUBSUB_KILL_COMMAND, (err, stdout) => {
+            if (err) {
+              this.logger.logLabeled("DEBUG", "pubsub", JSON.stringify(err));
+            }
+            if (stdout) {
+              this.logger.logLabeled("DEBUG", "pubsub", JSON.stringify(stdout));
+            }
+            resolve(null);
```

Comment:
Passing `null` isn't necessary.

```suggestion
            resolve();
```